### PR TITLE
Replace assert() with CHECK()

### PR DESCRIPTION
**Pull request by ArthurBandaryk**
On Linux target 'test:borrowed_ptr' doesn't build successfully
without 'cassert' header.

### CODE REVIEW

**Comment by ArthurBandaryk** [comment]
@onelxj suggested me to use `CHECK()` instead. I think it's a great idea , so I replaced all `assert()`s with `CHECK()`.

File: stout/borrowed_ptr.h [hunk 1]
```diff
@@ -47,7 +47,7 @@ class TypeErasedBorrowable {
         return true;
       }
 
-      assert(state == State::Borrowing);
+      CHECK(state == State::Borrowing);
```
Comment (benh) [inline stout/borrowed_ptr.h]: CHECK_EQ

File: stout/borrowed_ptr.h [hunk 2]
```diff
@@ -145,10 +145,10 @@ class TypeErasedBorrowable {
   void Reborrow() {
     auto [state, count] = tally_.Wait([](auto, size_t) { return true; });
 
-    assert(count > 0);
+    CHECK(count > 0);
```
Comment (benh) [inline stout/borrowed_ptr.h]: CHECK_GT

File: stout/borrowed_ptr.h [hunk 3]
```diff
@@ -145,10 +145,10 @@ class TypeErasedBorrowable {
...
 
     do {
-      assert(state != State::Destructing);
+      CHECK(state != State::Destructing);
```
Comment (benh) [inline stout/borrowed_ptr.h]: CHECK_NE

**Review by benh — APPROVED** [review]

File: stout/borrowed_ptr.h [hunk 4]
```diff
@@ -145,16 +153,33 @@ class TypeErasedBorrowable {
...
+    const TypeErasedBorrowable::State state) {
+  switch (state) {
+    case TypeErasedBorrowable::State::Borrowing:
+      return os << "Borrowing state";
+    case TypeErasedBorrowable::State::Watching:
+      return os << "Watching state";
+    case TypeErasedBorrowable::State::Destructing:
+      return os << "Destructing state";
+    default:
+      return os << "Unknown state";
```
Comment (onelxj) [inline stout/borrowed_ptr.h]: LOG(FATAL) << "unreachable";

**Review by onelxj — APPROVED** [review]

File: stout/borrowed_ptr.h [hunk 5]
```diff
@@ -145,16 +151,33 @@ class TypeErasedBorrowable {
...
     do {
-      assert(state != State::Destructing);
+      CHECK_NE(state, State::Destructing);
     } while (!tally_.Increment(state));
   }
 };
 
 ////////////////////////////////////////////////////////////////////////
 
+std::ostream& operator<<(
```
Comment (benh) [inline stout/borrowed_ptr.h]: Let's just put the implementation where we define the friend above!

**Review by benh — APPROVED** [review]

**Pull request merged**